### PR TITLE
Fix data race

### PR DESCRIPTION
**Pull request by Honglei-Cong**

### CODE REVIEW

File: consensus/vbft/block_pool.go [hunk 1]
```diff
@@ -631,15 +631,22 @@ func (pool *BlockPool) addSignaturesToBlockLocked(block *Block, forEmpty bool) e
 			}
 		}
 	}
+
+	// clone a new block
+	newBlk := &Block{}
+	if err := newBlk.Deserialize(block.Serialize()); err != nil {
```
Comment (laizy) [inline consensus/vbft/block_pool.go]: this looks too hacky, adding a new proposalSig field in blockProposalMsg to avoid data race.

Comment (Honglei-Cong) [reply consensus/vbft/block_pool.go]: Right.  But block signatures needs to be updated, and there'll be data race anyway.
You'll have to add lock to block header, or copy a new one.

Comment (laizy) [reply consensus/vbft/block_pool.go]: this update function is called when the blockPool's lock holds, I do not expect there is any data race left.

Comment (Honglei-Cong) [reply consensus/vbft/block_pool.go]: No lock when reading signatures in block header, but signature array in block header must be updated.
Either adding locks for all read operations, or just clone a new one when updating then there'll no write ops to block header?

Comment (Honglei-Cong) [reply consensus/vbft/block_pool.go]: blockpool's lock can not protect block headers in proposal message?  

ヽ༼ຈʖ̯ຈ༽ﾉ
--

Comment (Honglei-Cong) [reply consensus/vbft/block_pool.go]: updated

**Review by laizy — APPROVED** [review]

**Pull request merged**